framework/src/consumption/lib/quicksight/quicksight-subscription.ts

```diff
@@ -0,0 +1,232 @@
+   */
+  readonly adminGroup: string[];
+
+  /**
+   * The author group associated with your IAM Identity Center account.
+   */
+  readonly authorGroup: string[];
+
+  /**
+   * The reader group associated with your IAM Identity Center account.
+   */
+  readonly readerGroup: string[];
+
+  /**
+  * The region to use as main QuickSight region (used to store configuration and identities info)
+  */
+  readonly identityRegion: string;
+
+  private readonly removalPolicy: RemovalPolicy;
+
+  private readonly serviceToken: string;
+  private readonly policyActions: string[];
+
+  constructor (scope: Construct, id: string, props: QuickSightSubscriptionProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: QuickSightSubscription.name,
+    };
+    super(scope, id, trackedConstructProps);
+
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+    this.accountName = props.accountName;
+    this.notificationEmail = props.notificationEmail;
+    this.adminGroup = props.adminGroup;
+    this.authorGroup = props.authorGroup;
+    this.readerGroup = props.readerGroup;
+    this.identityRegion = props.identityRegion;
+
+    this.policyActions = [
+      'quicksight:Subscribe',
+      'quicksight:UpdateAccountSettings',
+      'quicksight:Create*',
+      'quicksight:Unsubscribe',
+      'quicksight:DescribeAccountSubscription',
+      'sso:GetManagedApplicationInstance',
+      'sso:CreateManagedApplicationInstance',
+      'sso:GetManagedApplicationInstance',
+      'sso:DeleteManagedApplicationInstance',
+      'sso:GetManagedApplicationInstance',
+      'sso:DescribeGroup',
+      'sso:SearchGroups',
+      'sso:GetProfile',
+      'sso:AssociateProfile',
+      'sso:DisassociateProfile',
+      'sso:ListProfiles',
+      'sso:ListDirectoryAssociations',
+      'sso:DescribeRegisteredRegions',
+    ];
+
+    if (props.authenticationMethod != 'IAM_IDENTITY_CENTER') {
+      this.policyActions = this.policyActions.concat(
+        [
+          'ds:AuthorizeApplication',
+          'ds:UnauthorizeApplication',
+          'ds:CheckAlias',
+          'ds:CreateAlias',
+          'ds:DescribeDirectories',
+          'ds:DescribeTrusts',
+          'ds:DeleteDirectory',
+          'ds:CreateIdentityPoolDirectory',
+        ],
+      );
+    }
+
+    this.executionRole = new Role(this, 'Role', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaVPCAccessExecutionRole'),
+      ],
+      inlinePolicies: {
+        QuickSightSubscription: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: this.policyActions,
+              resources: ['*'],
+            }),
+          ],
+        }),
+      },
+    });
+
+    const timeout = props.executionTimeout || Duration.minutes(5);
```

Comment:
can you make this check `null` and `not false`. meaning ?? instead of ||.

Reply:
Ack.